pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -543,6 +552,91 @@ func (a *Actuator) setHostSpec(ctx context.Context, host *bmh.BareMetalHost, mac
 	return a.client.Update(ctx, host)
 }
 
+// updateUserData fetches full ignition from MCS and updates the UserData secret reference
+func (a *Actuator) updateUserData(ctx context.Context, bmconfig *bmv1alpha1.BareMetalMachineProviderSpec,
+	machine *machinev1beta1.Machine) error {
+
+	// Try to fetch the full ignition secret to see if it exists
+	machineUserDataSecret := &corev1.Secret{}
+	key := client.ObjectKey{
+		Name:      machine.Name + "-user-data",
+		Namespace: machine.Namespace,
+	}
+	// Check if the full ignition secret already exists. If not, create one
+	err := a.client.Get(ctx, key, machineUserDataSecret)
+	if err == nil {
+		return nil
+	}
+	if !errors.IsNotFound(err) {
+		log.Printf("Cannot get %s secret: %s", machineUserDataSecret.Name, err.Error())
+		return fmt.Errorf("Cannot get %s secret: %w", machineUserDataSecret.Name, err)
+	}
+
+	// Get the worker machineconfigpool
+	mcp := &mcfgv1.MachineConfigPool{}
+	key = client.ObjectKey{
+		Name: "worker",
```

Comment:
We can't hard-code this, it'll need to be derived from the machine role to support custom machinesets

Reply:
+1. This is now fixed. Thanks for catching this one.